### ISSUE TITLE
Lint: result should be output to stdout instead of stderr

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno#7304]
Currently, the result of `deno lint`is output to _stderr_ as follows

```
 $ deno lint --unstable 1>/dev/null
(no-explicit-any) `any` type is not allowed
let a: any = 1;
       ~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/tmp.ts:1:7

Found 1 problem
```

ESLint, on the other hand, outputs a result to _stdout_. You can see

```
$ npx eslint tmp.ts 2>/dev/null

/home/yusuktan/Repos/github.com/magurotuna/deno/tmp.ts
  1:5  error    'a' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  1:5  error    'a' is never reassigned. Use 'const' instead                                  prefer-const
  1:8  warning  Unexpected any. Specify a different type                                      @typescript-eslint/no-explicit-any

✖ 3 problems (2 errors, 1 warning)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

I think ESLint's way is preferable and `deno lint` also should output its result to stdout.

Is there any specific reason outputting to stderr, which I haven't noticed?

**Comment by bartlomieju** [comment]
Any output from linter is considered as error, so I guess it should go to stderr? 

> I think ESLint's way is preferable and deno lint also should output its result to stdout.

@magurotuna what's the reason to prefer stdout?

**Comment by nayeemrmn** [comment]
Duplicate of #6303. See https://github.com/denoland/deno/issues/6303#issuecomment-644316711. Lint outputs nothing on success, and diagnostics on failure.

**Comment by magurotuna** [comment]
@nayeemrmn 
Sorry I didn't notice the issue. Thanks for the information.

@bartlomieju 
The reason why I think going to stdout would be better is that we expect the linter to report diagnostics. In other words,
- stderr should be used to output a report about panics happening while linting. For example, in case that the parser cannot parse a script because it has broken code, then its report should go to stderr rather than stdout.
- stdout should be used to output diagnostics of lint. Printing lint diagnostics is not an error of the linter itself.

Of course, the fact that ESLint's result is output to stdout is also one of the reason.

But if there is already a consensus among the Deno community, I think it's okay.

**Comment by nayeemrmn** [comment]
> The reason why I think going to stdout would be better is that we expect the linter to report diagnostics. In other words,
> 
> * stderr should be used to output a report about panics happening while linting. For example, in case that the parser cannot parse a script because it has broken code, then its report should go to stderr rather than stdout.
> * stdout should be used to output diagnostics of lint. Printing lint diagnostics is not an error of the linter itself.

So it should exit with code 0 unless a panic happens, yes? No, we exit with code 1 when there are lint errors so by the same logic those lint errors should be written to stderr.

> we expect the linter to report diagnostics

Again, this is false. We expect it to print nothing. Diagnostics are unexpected.

**Comment by magurotuna** [comment]
@nayeemrmn 
I understand. As you say, exit code is 1 when there are lint errors, which I forgot to consider. And I find most lint tools other than ESLint output their lint errors to stderr. This fact supports your opinion.

Then, where do you think is it suitable for information to be output, such as how many files have been checked by the linter? It should be written to stdout?
I have implemented this feature and opened a PR (#7312). In current implementation `Checked x files` goes to stderr; I wonder if it's okay.
